docusaurus/docs/create-a-plugin/extend-a-plugin/nested-plugins.md

```diff
@@ -0,0 +1,87 @@
+---
+id: nested-plugins
+title: Work with nested plugins
+description: How to work with nested plugins, Grafana app plugins.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - advanced
+  - apps
+  - app plugins
+  - nested
+---
+
+import ScaffoldNPM from '@snippets/createplugin-scaffold.npm.md';
+import ScaffoldPNPM from '@snippets/createplugin-scaffold.pnpm.md';
+import ScaffoldYarn from '@snippets/createplugin-scaffold.yarn.md';
+
+Grafana app plugins can nest data sources, both frontend and backend, together with panel plugins so that you can provide a complete user experience.
+
+## Before you begin
+
+See the tutorial to [build an app plugin](../../tutorials/build-an-app-plugin) and scaffold an app plugin.
+
+## Anatomy of nested plugins
+
+Nested plugins are located inside the app plugin `src` folder. They generally follow the same structure of a plugin and have their own `plugin.json`, but they don't have their own `package.json` or `.config` folder.
+
+Here's an example of a nested data source plugin:
+
+```diff bash
+./src
+ ├── README.md
+ ├── components
++├── nested-datasource
++│   ├── components
++│   │   ├── ConfigEditor.tsx
++│   │   └── QueryEditor.tsx
++│   ├── datasource.ts
++│   ├── img
++│   ├── module.ts
++│   ├── plugin.json
++│   └── types.ts
+ ├── img
+ │   └── logo.svg
+ ├── module.ts
+ └── plugin.json
+```
+
+:::note
+
+Notice that the plugin `nested-datasource` doesn't have a `package.json` of its own. The name of the nested plugin folder is not important.
+
+:::
+
+## How to add a nested plugin to an app plugin
+
+1. Generate the nested plugin:
+
+   :::important
+
+   Begin outside your app plugin's directory.
+
+   :::
+
+   Use the `create-plugin` tool to generate a new plugin:
+
+   ```bash
+   npx @grafana/create-plugin@latest
+   ```
+
+   Carefully select the desired plugin type (data source or panel), provide a name, and use the same organization as your app plugin.
+
+1. Prepare the nested plugin:
+
+   Rename the `src` folder of your newly generated plugin to a name that reflects its specific purpose (for example, `nested-datasource`).
+
+1. Integrate into your app plugin:
+
+   Copy the renamed `src` folder directly inside the `src` folder of your app plugin.
+   You can safely disregard the other generated files (like `package.json`, `.config`, and so on). These files aren't needed within your app.
+
+1. (Optional) Clean up your directory:
+
+   You can now delete the entire directory of the initially generated nested plugin.
+
+With these steps, your app plugin now houses the source code for your nested data source or panel, ready for further development.
```

Comment:
Are there further steps needed here to be able to use the new nested plugins, such as updating the `GF_LOG_FILTERS` and `GF_PLUGINS_ALLOW_LOADING_UNSIGNED_PLUGINS` environment variables in the `docker-compose.yaml` file?

We should probably also include instructions for updating the `provisioning/datasources/datasources.yml` file if the user was adding a new nested datasource plugin.

Reply:
There's no need to add the nested plugins to the allow unsigned. 

>We should probably also include instructions for updating the provisioning/datasources/datasources.yml file if the user was adding a new nested datasource plugin.

That is covered in other parts of the documentation. The process is the same.

Reply:
> That is covered in other parts of the documentation. The process is the same.

I think a simple call out to the other docs page as a reminder would go a long way here, as a developer may not be aware that it's a step they need to take. This somewhat relates to the frustration I had when navigating the docs, there's an expectation of implied knowledge that the developer has already read other docs pages (or even knows they exist).

Reply:
that sounds good.